data-prepper-core/src/test/java/com/amazon/dataprepper/parser/config/MetricsConfigTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.parser.config;
+
+import com.amazon.dataprepper.parser.model.DataPrepperConfiguration;
+import com.amazon.dataprepper.parser.model.MetricRegistryType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.isA;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+class MetricsConfigTest {
```

Comment:
Do we need test to verify bean creation ? We can probably exclude the configuration file from test coverage if the intension was to have code coverage for the bean creation class.

Reply:
The intention is only to provide coverage to the configuration class. While, this one it very simple and unit tests is excessive. Other configuration classes hold complex logic, and I did not want to set the standard for excluding them. In the next PR, this class will increase in complexity and will require testing.

@cmanning09 had similar comments on these tests. If we exclude them a standard will need a standard to prevent excluding coverage where needed.